pkg/util.go

```diff
@@ -129,3 +179,568 @@ func (opt *etcdOptions) setCredentials(sh Shell, appBinding *appcatalog.AppBindi
 	sh.SetEnv(EnvEtcdPassword, string(secret.Data[EtcdPassword]))
 	return nil
 }
+
+func clearDir(dir string) error {
+	if err := os.RemoveAll(dir); err != nil {
+		return fmt.Errorf("unable to clean datadir: %v. Reason: %v", dir, err)
+	}
+	return os.MkdirAll(dir, os.ModePerm)
+}
+
+func (opt *etcdOptions) createRestorePvcs(pvcList []corev1.PersistentVolumeClaim) ([]corev1.PersistentVolumeClaim, error) {
+
+	if len(pvcList) == 0 {
+		err := errors.New("No pvc found for etcd statefulset")
+		return nil, err
+	}
+
+	restorePvcList := make([]corev1.PersistentVolumeClaim, 0)
+	inv, err := invoker.ExtractRestoreInvokerInfo(opt.kubeClient, opt.stashClient, opt.InvokerKind, opt.InvokerName, opt.namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceRequirement := pvcList[0].Spec.Resources
+
+	if opt.memReq != "" || opt.memLimit != "" {
+		resourceRequirement = corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse(opt.cpuLimit),
+				corev1.ResourceMemory: resource.MustParse(opt.memLimit),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse(opt.cpuReq),
+				corev1.ResourceMemory: resource.MustParse(opt.memReq),
+			},
+		}
+	}
+	if opt.storageClassName == "" {
+		opt.storageClassName = *pvcList[0].Spec.StorageClassName
+	}
+
+	for _, pvcs := range pvcList {
```

Comment:
Don't use pre-calculated PVC list. Add `restoreMember()` function where you will send member pod as argument. That function will create the PVC and run download into it.